src/events/events.ts

```diff
@@ -0,0 +1,64 @@
+import { OnAppClose } from '../hook/interfaces.ts';
+import { Logger, Module } from '../mod.ts';
+
+// deno-lint-ignore no-explicit-any
+type Listener<P = any> = (payload: P) => void;
+
+export class EventEmitter implements OnAppClose {
+	private logger: Logger = new Logger('EventEmitter');
+	private listenersRegistered: Array<[string, Listener]>;
```

Comment:
I'm keeping this array to allow us to `unsubscribe` from one (or all) channel while keeping the `EventTarget` instance to be reused

Reply:
We can probably optimize that by using a map instead of an array.

Such as:

`private listenersRegistered: Map<string, Listener[]>;`

So when you want to get listeners for a given you can do:

`listenersRegistered.get(channel);`

And when you want to delete listeners for a channel:

`listenersRegistered.delete(channel)`

To get all listeners for all channels:

`Array.from(listenerRegistered.entries()).flatMap()`

Reply:
Thanks @Sorikairox!

I did the unsubscribe all a little different, lmk what you think